src/Controller/TranslatorController.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2020 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+namespace App\Controller;
+
+/**
+ * Translator controller.
+ *
+ * @property \App\Controller\Component\TranslatorComponent $Translator
+ */
+class TranslatorController extends AppController
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function initialize(): void
+    {
+        parent::initialize();
+        $this->loadComponent('Translator');
+        $this->Security->setConfig('unlockedActions', ['translate']);
+    }
+
+    /**
+     * Translate text.
+     *
+     * @return void
+     */
+    public function translate(): void
+    {
+        $this->viewBuilder()->setClassName('Json');
+        $this->request->allowMethod(['post']);
+        $text = $this->request->getData('text', '');
+        $texts = (is_array($text)) ? $text : [$text];
+        try {
+            $json = $this->Translator->translate(
+                $texts,
+                (string)$this->request->getData('from'),
+                (string)$this->request->getData('to')
+            );
+            $decoded = json_decode($json);
+            $this->set('translation', $decoded->translation);
+        } catch (\Exception $e) {
+            $error = $e->getMessage();
```

Comment:
An exception thrown here (for example in case of missing or wrong api token) will result in a javascript error

* translated field becomes blank => it should be unchanged
* a js error appears in console: 
```
SyntaxError: Unexpected token < in JSON at position 0 

autoTranslate @ app.bundle.12140b.js:1
async function (async)
autoTranslate @ app.bundle.12140b.js:1
fetchTranslation @ map-view.bundle.f1d9a6.js:1
translate @ map-view.bundle.f1d9a6.js:1
click @ VM671:3
Ue @ vendors.bundle.7f31bd.js:7
n @ vendors.bundle.7f31bd.js:7
Kr.o._wrapper @ vendors.bundle.7f31bd.js:7
```